streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java

```diff
@@ -350,9 +353,35 @@ public boolean isOpen() {
         return open;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public <R> QueryResult<R> query(final Query<R> query, final PositionBound positionBound,
         final boolean collectExecutionInfo) {
+
+        if (query instanceof WindowKeyQuery) {
+            final WindowKeyQuery<Bytes, byte[]> windowKeyQuery = (WindowKeyQuery<Bytes, byte[]>) query;
+            if (windowKeyQuery.getTimeFrom().isPresent() && windowKeyQuery.getTimeTo().isPresent()) {
+                final Bytes key = windowKeyQuery.getKey();
+                final Instant lower = windowKeyQuery.getTimeFrom().get();
+                final Instant upper = windowKeyQuery.getTimeTo().get();
+                final WindowStoreIterator<byte[]> iterator = this.fetch(key, lower, upper);
+                final R result = (R) iterator;
+                final QueryResult<R> queryResult = QueryResult.forResult(result);
+                return queryResult;
+            }
+        } else if (query instanceof WindowRangeQuery) {
+            final WindowRangeQuery<Bytes, byte[]> windowRangeQuery = (WindowRangeQuery<Bytes, byte[]>) query;
+            if (windowRangeQuery.getTimeFrom().isPresent() &&
+                    windowRangeQuery.getTimeTo().isPresent()) {
```

Comment:
It looks like this method will return "unknown query" if the window bounds aren't specified. Can we make sure there's a test case for `WindowRangeQuery#withKey`?